### ISSUE TITLE
minikube profile list --output json fails with ENONT after fresh install

### ISSUE THREAD

**Issue by nirs** [kubernetes/minikube#15593]
### What Happened?

After minikube on a fresh machine, running:

    minikube profile list --output json

Fails, writing error to stdout:

    {"error":{"Op":"open","Path":"/home/runner/.minikube/profiles","Err":2}}

The expected output:

    {"invalid":[],"valid":[]}

Issues:
- Missing ~/.minikube/profiles not handled gracefully
- Reporting errors to stdout
- Reporting error number (2) instead of description (No such file or directory).

Workaround:
- Create .minikube/profiles manually before running the command

### Attach the log file

Full reproducer:

Add this github workflow:
```
name: CI
on:
  - push
  - pull_request
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - name: Reproduce minikube bug
        run: |
          curl -LO https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64
          sudo install minikube-linux-amd64 /usr/local/bin/minikube
          minikube --alsologtostderr --v=7 profile list --output json
```

Example run:

```
I0104 21:22:57.498168    1632 out.go:296] Setting OutFile to fd 1 ...
I0104 21:22:57.498301    1632 out.go:343] TERM=,COLORTERM=, which probably does not support color
I0104 21:22:57.498306    1632 out.go:309] Setting ErrFile to fd 2...
I0104 21:22:57.498311    1632 out.go:343] TERM=,COLORTERM=, which probably does not support color
I0104 21:22:57.503171    1632 root.go:334] Updating PATH: /home/runner/.minikube/bin
W0104 21:22:57.503308    1632 root.go:311] Error reading config file at /home/runner/.minikube/config/config.json: open /home/runner/.minikube/config/config.json: no such file or directory
I0104 21:22:57.503638    1632 out.go:303] Setting JSON to true
I0104 21:22:57.503749    1632 notify.go:220] Checking for updates...
{"error":{"Op":"open","Path":"/home/runner/.minikube/profiles","Err":2}}
```

Can also be reproduced by removing (an empty) `~/.minikube/profiles` directory.

### Operating System

Ubuntu

### Driver

N/A

**Comment by nirs** [comment]
The same issue #5898 was reported and fixed #5955 in 2019.

**Comment by nirs** [comment]
This issue is addressed by #15594.
